vm/chef/cookbooks/c2d-config/files/c2d-utils

```diff
@@ -65,6 +65,17 @@ function has_external_ip() {
   fi
 }
 
+function wait_for_port() {
+  local -r host="$1"
+  local -r port="$2"
+  local -r timeout="${3:-300}"
+
+  timeout --preserve-status "${timeout}" bash -c "until echo > /dev/tcp/${host}/${port}; do sleep 2; done"
+  if [[ "$?" -ne 0 ]]; then
+    exit 1
```

Comment:
Is it better to return a value rather than make the script exit?

Reply:
Well, I just thought about making the usage easier. As: 

```shell
wait_for_port "elastic-vm-01" 9200

# keep script execution...
```

Instead of: 

```shell
result="$(wait_for_port elastic-vm-01 9200)"

if [[ "${result}" -eq .... 
```
or 
```shell
wait_for_port elastic-vm-01 9200

if [[ "$?" -eq .... 
```

@gibbleyg , what do you think?